server/channels/app/channel.go

```diff
@@ -705,7 +707,9 @@ func (a *App) UpdateChannelPrivacy(c request.CTX, oldChannel *model.Channel, use
 			channel.Type = model.ChannelTypeOpen
 		}
 		// revert to previous channel privacy
-		a.UpdateChannel(c, channel)
+		if _, err = a.UpdateChannel(c, channel); err != nil {
+			return nil, err
+		}
```

Comment:
This being already an error case (`postChannelPrivacyMessage` has failed), we could be losing useful information. I'd probably consider wrapping both errors. @hanzei thoughts?

Reply:
If there is a way to warp both errors into one ` *model.AppError`, I'm fully supportive.  Otherwise, we can just log this error and return the one from `postChannelPrivacyMessage`.

Reply:
Any conclusion?

Reply:
I think we should just log the error from `UpdateChannel`.  That is the way we do in the rest of the code base, when we need to revert to a state.

Reply:
I lost this branch! 🙃

Reply:
Did you lose the branch on your local machine? No, problem, you can still fetch it from GitHub. See https://github.com/RS-labhub/mattermost/tree/channel